tests/components/demo/test_vacuum.py

```diff
@@ -47,326 +47,293 @@
+    assert state.attributes.get(ATTR_FAN_SPEED_LIST) is None
+    assert state.state == STATE_OFF
+
+    state = hass.states.get(ENTITY_VACUUM_NONE)
+    assert state.attributes.get(ATTR_SUPPORTED_FEATURES) == 0
+    assert state.attributes.get(ATTR_STATUS) is None
+    assert state.attributes.get(ATTR_BATTERY_LEVEL) is None
+    assert state.attributes.get(ATTR_FAN_SPEED) is None
+    assert state.attributes.get(ATTR_FAN_SPEED_LIST) is None
+    assert state.state == STATE_OFF
+
+    state = hass.states.get(ENTITY_VACUUM_STATE)
+    assert state.attributes.get(ATTR_SUPPORTED_FEATURES) == 13436
+    assert state.state == STATE_DOCKED
+    assert state.attributes.get(ATTR_BATTERY_LEVEL) == 100
+    assert state.attributes.get(ATTR_FAN_SPEED) == "medium"
+    assert state.attributes.get(ATTR_FAN_SPEED_LIST) == FAN_SPEEDS
+
+
+async def test_methods(hass):
+    """Test if methods call the services as expected."""
+    hass.states.async_set(ENTITY_VACUUM_BASIC, STATE_ON)
+    await hass.async_block_till_done()
+    assert vacuum.is_on(hass, ENTITY_VACUUM_BASIC)
+
+    hass.states.async_set(ENTITY_VACUUM_BASIC, STATE_OFF)
+    await hass.async_block_till_done()
+    assert not vacuum.is_on(hass, ENTITY_VACUUM_BASIC)
+
+    await common.async_turn_on(hass, ENTITY_VACUUM_COMPLETE)
+    assert vacuum.is_on(hass, ENTITY_VACUUM_COMPLETE)
+
+    await common.async_turn_off(hass, ENTITY_VACUUM_COMPLETE)
+    assert not vacuum.is_on(hass, ENTITY_VACUUM_COMPLETE)
+
+    await common.async_toggle(hass, ENTITY_VACUUM_COMPLETE)
+    assert vacuum.is_on(hass, ENTITY_VACUUM_COMPLETE)
+
+    await common.async_start_pause(hass, ENTITY_VACUUM_COMPLETE)
+    assert not vacuum.is_on(hass, ENTITY_VACUUM_COMPLETE)
+
+    await common.async_start_pause(hass, ENTITY_VACUUM_COMPLETE)
+    assert vacuum.is_on(hass, ENTITY_VACUUM_COMPLETE)
+
+    await common.async_stop(hass, ENTITY_VACUUM_COMPLETE)
+    assert not vacuum.is_on(hass, ENTITY_VACUUM_COMPLETE)
+
+    state = hass.states.get(ENTITY_VACUUM_COMPLETE)
+    assert state.attributes.get(ATTR_BATTERY_LEVEL) < 100
+    assert state.attributes.get(ATTR_STATUS) != "Charging"
+
+    await common.async_locate(hass, ENTITY_VACUUM_COMPLETE)
+    state = hass.states.get(ENTITY_VACUUM_COMPLETE)
+    assert "I'm over here" in state.attributes.get(ATTR_STATUS)
+
+    await common.async_return_to_base(hass, ENTITY_VACUUM_COMPLETE)
+    state = hass.states.get(ENTITY_VACUUM_COMPLETE)
+    assert "Returning home" in state.attributes.get(ATTR_STATUS)
+
+    await common.async_set_fan_speed(
+        hass, FAN_SPEEDS[-1], entity_id=ENTITY_VACUUM_COMPLETE
+    )
+    state = hass.states.get(ENTITY_VACUUM_COMPLETE)
+    assert state.attributes.get(ATTR_FAN_SPEED) == FAN_SPEEDS[-1]
+
+    await common.async_clean_spot(hass, entity_id=ENTITY_VACUUM_COMPLETE)
+    state = hass.states.get(ENTITY_VACUUM_COMPLETE)
+    assert "spot" in state.attributes.get(ATTR_STATUS)
+    assert state.state == STATE_ON
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5315232)